lib/node_modules/@stdlib/lapack/base/dlatrs/lib/base.js

```diff
@@ -0,0 +1,522 @@
+* @param {Float64Array} CNORM - used to store the column norms
+* @param {integer} strideCNORM - stride length for `CNORM`
+* @param {NonNegativeInteger} offsetCNORM - starting index for `CNORM`
+* @returns {number} scaling factor
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 2.0, 1.0, -1.0, 0.0, 3.0, 2.0, 0.0, 0.0, 4.0 ] ); // => [ [ 2.0, 1.0, -1.0 ], [ 0.0, 3.0, 2.0 ], [ 0.0, 0.0, 4.0 ] ]
+* var X = new Float64Array( [ 5.0, 10.0, 20,0 ] );
+* var CNORM = new Float64Array( 3 );
+*
+* var scale = dlatrs( 'upper', 'no-transpose', 'non-unit', 'no', 3, A, 3, 1, 0, X, 1, 0, CNORM, 1, 0 );
+* // returns 1.0
+* // X => <Float64Array>[ 5.0, 0.0, 5.0 ]
+* // CNORM => <Float64Array>[ 0.0, 1.0, 3.0 ]
+*/
+function dlatrs( uplo, trans, diag, normin, N, A, strideA1, strideA2, offsetA, X, strideX, offsetX, CNORM, strideCNORM, offsetCNORM ) {
+	var jfirst;
+	var scale;
+	var jlast;
+	var tscal;
+	var uscal;
+	var GOTO;
+	var jinc;
+	var imax;
+	var tmax;
+	var xmax;
+	var grow;
+	var xbnd;
+	var tjjs;
+	var sumj;
+	var tjj;
+	var rec;
+	var ix1;
+	var ia1;
+	var ia2;
+	var ia;
+	var xj;
+	var ix;
+	var ic;
+	var j;
+	var i;
+
+	scale = 1.0;
+	if ( N === 0 ) {
+		return 1.0;
+	}
+
+	// Compute the 1 norm of each column excluding the diagonal elements
+	if ( normin === 'no' ) {
+		if ( uplo === 'upper' ) { // upper triangular
+			ic = offsetCNORM;
+			ia = offsetA;
+			for ( j = 0; j < N; j++ ) {
+				CNORM[ ic ] = dasum( j, A, strideA1, ia );
+				ia += strideA2;
+				ic += strideCNORM;
+			}
+		} else { // lower triangular
+			ic = offsetCNORM;
+			ia = offsetA + strideA1; // follows A( j+1, j )
+			for ( j = 0; j < N - 1; j++ ) {
+				CNORM[ ic ] = dasum( N - ( j + 1 ), A, strideA1, ia );
+				ic += strideCNORM;
+				ia += strideA1 + strideA2;
+			}
+			CNORM[ ic ] = 0.0;
+		}
+	}
+
+	imax = idamax( N, CNORM, strideCNORM, offsetCNORM );
+	tmax = CNORM[ offsetCNORM + ( imax * strideCNORM ) ];
+	if ( tmax <= bignum ) {
+		tscal = 1.0;
+	} else {
+		if ( tmax <= dlamch( 'overflow' ) ) {
+			tscal = 1.0 / ( smlnum * tmax );
+			dscal( N, tscal, CNORM, strideCNORM, offsetCNORM );
+		} else {
+			tmax = 0.0;
+			if ( uplo === 'upper' ) {
+				ia = offsetA + strideA2;
+				for ( j = 1; j < N; j++ ) {
+					tmax = max( dlange( 'max', j, 1, A, strideA1, strideA2, ia ), tmax );
+					ia += strideA2;
+				}
+			} else {
+				ia = offsetA + strideA1;
+				for ( j = 0; j < N - 1; j++ ) {
+					tmax = max( dlange( 'max', N - ( j + 1 ), 1, A, strideA1, strideA2, ia ), tmax );
+					ia += strideA1 + strideA2;
+				}
+			}
+
+			if ( tmax <= dlamch( 'overflow' ) ) {
+				tscal = 1.0 / ( smlnum * tmax );
+				ic = offsetCNORM;
+				ia2 = offsetA;
+				for ( j = 0; j < N; j++ ) {
+					if ( CNORM[ ic ] <= dlamch( 'overflow' ) ) {
+						CNORM[ ic ] *= tscal;
+					} else {
+						CNORM[ ic ] = 0.0;
+						if ( uplo === 'upper' ) {
+							ia1 = 0;
+							for ( i = 0; i < j; i++ ) {
```

Comment:
```suggestion
							for ( i = 0; i < j-1; i++ ) {
```

Should this be `j-1`?

Reply:
i think so. yes

Reply:
@aayush0325 Does this change still need to be made?